### PR TITLE
pbench-run-benchmark linpack support

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Comment by k-rister** [comment]
DO NOT MERGE, not yet complete.

**Comment by k-rister** [comment]
This PR is focused on add support for running linpack via pbench-run-benchmark.  During the course of implementation several minor issues with the pbench-run-benchmark infrastructure have been encountered and fixes for those are included.

**Comment by k-rister** [comment]
Most notably, there is no postprocessing yet for the linpack results which still needs to be done.

**Comment by k-rister** [comment]
I think this is good to go now, the CI failures are something weird with the dashboard which this doesn't touch.

**Comment by portante** [comment]
@k-rister, can you rebase this patch set and squash them down to a smaller number of related commits?

Also PR #1490 seems to have fixed the failing CI tests for the most part, so a rebase would be great.  I'll review in the mean time.

File: agent/bench-scripts/pbench-run-benchmark [hunk 1]
```diff
@@ -360,8 +369,11 @@ while (scalar @param_sets > 0) {
         $iteration_id++;
     }
     if ($pp_only) {
-        print BULK_FH "echo Waiting for $nr_pp_jobs post-processing jobs to finish\n";
-        print BULK_FH "wait\n";
+	if ($params{'postprocess-mode'} eq 'cdm') {
```
Comment (portante) [inline agent/bench-scripts/pbench-run-benchmark]: Seems like a tabs vs spaces formatting disagreement here.

File: agent/bench-scripts/pbench-run-benchmark-sample [hunk 2]
```diff
@@ -216,12 +220,15 @@ if (exists $specs{"controller"}{"post-script"} and $specs{"controller"}{"post-sc
...
     printf { $fh_cmd } "%s\n", $postprocess_cmd;
     close($fh_cmd);
-    my $postprocess_output = `. $pp_cmd_file`;
-    my $rc = $?;
-    open(my $fh_output, ">" . $sample_dir . "/postprocess-output.txt");
-    printf { $fh_output } "%s\n", $postprocess_output;
+    my $postprocess_output = `. $pp_cmd_file 2>&1`;
+    my $rc = $? >> 8;
+    if (open(my $fh_output, ">" . $sample_dir . "/postprocess-output.txt")) {
+	printf { $fh_output } "%s\n\n", $postprocess_output;
```
Comment (portante) [inline agent/bench-scripts/pbench-run-benchmark-sample]: Seems like tabs vs spaces formatting issues here.

File: agent/bench-scripts/pbench-run-benchmark (hunk 1 above)
Comment (k-rister) [reply agent/bench-scripts/pbench-run-benchmark]: I'm not an expert in this area...but isn't this type of editor behavior supposed to be controlled by these lines at the top of the files:

```
#!/usr/bin/perl
# -*- mode: perl; indent-tabs-mode: t; perl-indent-level: 4 -*-
# vim: autoindent tabstop=4 shiftwidth=4 expandtab softtabstop=4 filetype=perl
```

If so, does that mean these not correct?

File: agent/bench-scripts/pbench-run-benchmark-sample (hunk 2 above)
Comment (k-rister) [reply agent/bench-scripts/pbench-run-benchmark-sample]: See above...

**Comment by k-rister** [comment]
I rebased the code and squashed the linpack related commits together.  I left the bug fixes as individual commits since that seems appropriate.

File: agent/bench-scripts/pbench-run-benchmark (hunk 1 above)
Comment (portante) [reply agent/bench-scripts/pbench-run-benchmark]: It looks like we have a fight going on with tabs and spaces, where we have tabs in the file, but have `expandtab` in the `vim` settings, but don't have that specified in the `emacs` settings.

I have opened issue #1500 to propose a path forward to fix this.

File: agent/bench-scripts/pbench-run-benchmark-sample (hunk 2 above)
Comment (portante) [reply agent/bench-scripts/pbench-run-benchmark-sample]: See #1500.

**Pull request merged**